### PR TITLE
Fix speaker registration issues

### PR DESCRIPTION
**Pull request by bear454**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md).
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works(if appropriate).
- [X] I have added necessary documentation (if appropriate).

#### Changes proposed in this pull request:

* A confirmed speaker cannot register if registration is not open; registration must be open to register. Period.
* Ticket generation doesn't depend on local file system, but uses carrierwave formats instead.

#### Fixes Issue

* #1827 
* #1828

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -66,7 +66,7 @@ def signed_in(user)
 
     can [:new, :create], Registration do |registration|
       conference = registration.conference
-      conference.registration_open? && !conference.registration_limit_exceeded? || conference.program.speakers.confirmed.include?(user)
+      conference.registration_open? && (!conference.registration_limit_exceeded? || conference.program.speakers.confirmed.include?(user))
```
Comment (Ana06) [inline app/models/ability.rb]: As I just wrote in https://github.com/openSUSE/osem/issues/1827#issuecomment-348426404, this is made in purpose. The expected condition is the one we already have:

``` Ruby
(conference.registration_open? && !conference.registration_limit_exceeded?) || conference.program.speakers.confirmed.include?(user)
```

Please remove the first commit 😄

**Comment by differentreality** [comment]
Registration thingies look OK, I also agree with the approach of 

Allow confirmed speakers to register for the conference:
  - if the conference accepts registrations (otherwise don't show the link in proposals tooltip)
  - even if the dates for registration have passed
  - even if the registration limit has exceeded (because well, the speakers will be there anyway)

@namangupta01 has also been working on this in https://github.com/openSUSE/osem/pull/1629
@bear454 would you like to take a look at that PR and perhaps suggest the changes there so that his PR can be merged?
Then this PR can be about tickets and perhaps the addition of some extra tests for registration.

**Comment by bear454** [comment]
okay, I'll withdraw that commit from this PR, so I can at least get the other part fixed. :/

**Comment by bear454** [comment]
Based on changes to #1629 , I removed the permissions commit from this PR. @Ana06 please update your review.

**Pull request merged**